TeachingRecordSystem/src/TeachingRecordSystem.Core/Services/Files/BlobStorageFileService.cs

```diff
@@ -0,0 +1,60 @@
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using Azure.Storage.Sas;
+
+namespace TeachingRecordSystem.Core.Services.Files;
+
+public class BlobStorageFileService : IFileService
+{
+    private const string UploadsContainerName = "uploads";
+    private readonly BlobServiceClient _blobServiceClient;
+
+    public BlobStorageFileService(BlobServiceClient blobServiceClient)
+    {
+        _blobServiceClient = blobServiceClient;
+    }
+
+    public async Task<Guid> UploadFile(Stream stream, string? contentType)
+    {
+        var fileId = Guid.NewGuid();
+        var blobClient = await GetBlobClient(fileId);
+
+        await blobClient.UploadAsync(stream, httpHeaders: !string.IsNullOrEmpty(contentType) ? new BlobHttpHeaders { ContentType = contentType } : null);
+        return fileId;
+    }
+
+    public async Task<string> GetSasConnectionString(Guid fileId, int minutes)
+    {
+        var blobClient = await GetBlobClient(fileId);
+
+        var sasBuilder = new BlobSasBuilder
+        {
+            BlobContainerName = UploadsContainerName,
+            BlobName = fileId.ToString(),
+            ExpiresOn = DateTimeOffset.UtcNow.AddMinutes(minutes),
+            Protocol = SasProtocol.Https
+        };
+
+        sasBuilder.SetPermissions(BlobSasPermissions.Read);
+        return blobClient.GenerateSasUri(sasBuilder).ToString();
+    }
+
+    public async Task DeleteFile(Guid fileId)
+    {
+        var blobClient = await GetBlobClient(fileId);
+        await blobClient.DeleteAsync(DeleteSnapshotsOption.IncludeSnapshots);
+    }
+
+    private async Task<BlobClient> GetBlobClient(Guid fileId)
+    {
+        var blobContainerClient = await GetBlobContainerClient();
+        return blobContainerClient.GetBlobClient(fileId.ToString());
+    }
+
+    private async Task<BlobContainerClient> GetBlobContainerClient()
+    {
+        var blobContainerClient = _blobServiceClient.GetBlobContainerClient(UploadsContainerName);
+        await blobContainerClient.CreateIfNotExistsAsync();
+        return blobContainerClient;
```

Comment:
Can we cache the result of this (in a field)? It's an additional redundant API call for every method otherwise.

Reply:
Maybe `Task<BlobContainerClient> EnsureBlobContainerClient()`?

Reply:
Ha only just saw the last bit but I did create an Ensure .. method